src/Database/ImgIndex.php

```diff
@@ -770,9 +770,20 @@ public function getDataRange($start, $end, $sourceId, $switchSources = false) {
             return false;
         }
 
+        // Limit the number of rows we return to a max of 15,000
+        // by taking every Nth row where where N is the stride
+        // needed to reduce the result to 15000 rows.
+        $row_limit = HV_MAX_ROW_LIMIT;
+        $num_rows = $result->num_rows;
+        $stride = ceil($num_rows / $row_limit);
+        $row_index = 0;
+
         $result_array = Array();
         while ( $row = $result->fetch_array(MYSQLI_ASSOC) ) {
-            array_push($data, $row);
+            if (($row_index % $stride) == 0) {
```

Comment:
The solution has to be in SQL. Further testing shows that this will still fail if dates in the query are extended.

This patch works for querying 1 month, but not 6 months.

Reply:
this code can be removed now.